feat: Поддержка запросов вещей

nsu12 opened the pull request.

Commit:
fix: добавлен функционал запросов вещей

Commit:
feat: добавлена поддержка пагинации

Commit:
fix: codestyle

Commit:
feat: тесты для REST-эндпоинтов

Commit:
fix: исправлена ошибка в тесте

Commit:
feat: дописана часть юнит тестов на сервисы

Commit:
feat: дописаны юнит тесты

Commit:
feat: добавлены интеграционные тесты

Umka7389 commented on src/test/java/ru/practicum/shareit/user/UserServiceImplTest.java:
```diff
@@ -0,0 +1,149 @@
+        user.setEmail(email);
+        return user;
+    }
+}
```
Не увидел тестов с использованием DataJpaTest  и JsonTest

nsu12 replied:
Согласен, как-то упустил эти тесты. Добавил те, которые просят по заданию

Umka7389 reviewed (commented):
Молодец, отличная работа! Замечания в комментариях

Commit:
feat: добавлены DataJpa и Json тесты

для jpa репозиториев содержащих методы с Query аннотацией добавлены DataJpa тесты для их покрытия;
также добавлен тест json сериализации для pojo объекта ErrorResponse в котором используется jackson аннотация.

Umka7389 reviewed (commented):
Молодец, отличная работа!
